JavaScript/JavaScriptSDK/Context/Operation.ts

```diff
@@ -12,7 +12,10 @@ module Microsoft.ApplicationInsights.Context {
         public syntheticSource: string;
 
         constructor() {
-            this.id = Util.newGuid();
+            this.id = Util.newGuid();            
+            if (window && window.location && window.location.pathname) {
+                this.name = window.location.pathname;
```

Comment:
My point was that you may use whatever customer passed in the initial trackPage() as a name, not read it from document.url. As I mentioned [here](http://apmtips.com/blog/2015/08/15/disable-browser-telemetry-from-the-page/) customer may choose not to send URL by changing 

``` javascript
appInsights.trackPageView();
```

which is ultimately just

``` javascript
appInsights.trackPageView(window.location.pathname, window.location.href);
```

to: 

``` javascript
appInsights.trackPageView("something else", window.location.href);
```
And customer would expect to see the same ```something else``` as operation name for ajax.